src/FileInput/FileInput.jsx

```diff
@@ -0,0 +1,142 @@
+      </g>
+    </g>
+  </svg>);
+}
+
+UploadingIcon.propTypes = {
+  percent: React.PropTypes.number.isRequired,
+};
+
+export class FileInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filename: null,
+      error: null,
+      progress: null,
+      success: null,
+    };
+    filepicker.setKey(props.filePickerKey);
+    this.onDropRejected = this.onDropRejected.bind(this);
+    this.onDropAccepted = this.onDropAccepted.bind(this);
+  }
+
+  onDropAccepted(files) {
+    const fileKey = Math.random();
+    this.setState({fileKey});
+    this.setState({filename: files[0].name, error: null, progress: 0, success: null});
+    filepicker.store(files[0],
+      (file) => {
+        if (this.state.fileKey !== fileKey) {
+          return; // If the user drops multiple files we want to only treat the last one as "success"
+        }
+        this.setState({success: true});
+        this.props.onChange(file.key);
+      },
+      () => { this.setState({error: "Unknown error"}); },
+      (progress) => {
+        if (this.state.fileKey !== fileKey) {
+          return; // If the user drops multiple files we want to only take progress from the active upload
+        }
+        // Reduce progress since filepicker hangs at 95% for awhile so we don't want to make it look fully uploaded
+        // when it's hanging.
+        this.setState({progress: progress * 0.8});
+      },
+    );
+  }
+
+  onDropRejected() {
+    this.setState({filename: null, error: `Invalid ${this.props.label}`, success: null});
+    this.props.onChange(null);
+  }
+
+  render() {
+    return (<Dropzone
+      accept={this.props.accept}
+      style={{}}
+      multiple={false}
+      onDropAccepted={this.onDropAccepted}
+      onDropRejected={this.onDropRejected}
+    >{({isDragActive, isDragReject}) => {
+      let message = `Upload ${this.props.label}`;
+      let icon = <DefaultIcon />;
+      let selected = false;
+      if (isDragActive) {
+        selected = true;
+        message = "";
+      } else if (isDragReject || this.state.error) {
+        selected = true;
+        message = this.state.error || `Invalid ${this.props.label}`;
+        icon = <ErrorIcon />;
+      } else if (this.state.success) {
+        selected = true;
+        message = this.state.filename;
+        icon = <SuccessIcon />;
+      } else if (this.state.filename) {
+        icon = <UploadingIcon percent={this.state.progress} />;
+        message = this.state.filename;
+        selected = true;
+      }
+      return (<FlexBox className="FileInput" grow >
```

Comment:
Ah I see what is happening.

![image](https://cloud.githubusercontent.com/assets/835698/23633318/28645f40-027a-11e7-8e2d-cf5c0432019e.png)

`display: none` makes the input element inaccessible and the clickable element is a div. It should most likely be a Button in this case.